packages/eslint-plugin-react-hooks/src/ReactiveDeps.js

```diff
@@ -0,0 +1,534 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/* eslint-disable no-for-of-loops/no-for-of-loops */
+
+'use strict';
+
+// -1 if not a reactive Hook.
+// 0 for useEffect/useMemo/useCallback.
+// 1 for useImperativeHandle.
+// For additionally configured Hooks, assume 0.
+function getReactiveHookCallbackIndex(node, options) {
+  if (
+    node.type === 'MemberExpression' &&
+    node.object.type === 'Identifier' &&
+    node.object.name === 'React' &&
+    node.property.type === 'Identifier' &&
+    !node.computed
+  ) {
+    return getReactiveHookCallbackIndex(node.property);
+  } else if (
+    node.type === 'Identifier' &&
+    (node.name === 'useEffect' ||
+      node.name === 'useLayoutEffect' ||
+      node.name === 'useCallback' ||
+      node.name === 'useMemo')
+  ) {
+    return 0;
+  } else if (
+    node.type === 'Identifier' &&
+    node.name === 'useImperativeHandle'
+  ) {
+    return 1;
+  } else if (options && options.additionalHooks) {
```

Comment:
How is this intended to be used? (When would you need it?)

Reply:
When you have your own Hooks with deps like `useMyCustomEffect` that you want to check. I'm open to removing this option (it was in first @calebmer's draft) but also seems like it isn't hard to support.

Reply:
A possible heuristic is to guess that hooks which receive a callback followed by an array as their last argument are following the useMemo pattern